bin/create-pr-from-issue.rb

```diff
@@ -0,0 +1,149 @@
+def return_insert(gh_user, perm, name, email, org, reason, added_by, review_after)
+    <<~EOF
+    \s\s\s\s{
+        \s\sgithub_user  = "#{gh_user}"
+        \s\spermission   = "#{perm}"
+        \s\sname         = "#{name}"         
+        \s\semail        = "#{email}"        
+        \s\sorg          = "#{org}"          
+        \s\sreason       = "#{reason}"       
+        \s\sadded_by     = "#{added_by}"     
+        \s\sreview_after = "#{review_after}" 
+    \s\s\s\s},
+    EOF
+end
+
+# Creates a branch and commits all changed files
+# This probably deserves its own class but keeping it here for now until we need more major functionality in this area
+def create_branch_for_pr
+    # Init local Git
+    g = Git.open(".")
+  
+    g.config("user.name", "Operations Engineering Bot")
+    g.config("user.email", "dummy@email.com")
+  
+    # Generate random uuid for branch name
+    branch_name = UUIDTools::UUID.timestamp_create.to_s
+  
+    # Create branch and checkout
+    g.branch(branch_name).create
+    g.checkout(branch_name)
+  
+    # Stage file
+    g.add(paths = "terraform/*")
+  
+    # Commit
+    g.commit("Pull request to add new collaborators")
+  
+    # Push
+    g.push(
+      remote = g.remote("origin"),
+      branch = branch_name
+    )
+  
+    # Cleanup
+    g.checkout("main")
+  
+    # Return branch name for PR creation
+    branch_name
+end
+
+# Returns string for PR body
+def pull_hash(branch)
+    {
+        title: "External Collaborator PR",
+        head: branch,
+        base: "main",
+        body: <<~EOF
+        Hi there
+        
+        Please merge this pull request to add the attached external collaborators to GitHub.
+        
+        If you have any questions, please post in #ask-operations-engineering on Slack.
+        EOF
+    }
+    end
+
+# Util function to deal with Terraform . limitations
+def tf_safe(string)
+    string.tr(".", "-").strip
+end
+
+# Creates a hash of arrays with field: [0] value
+# From a GitHub issues reponse created from an issue template
+# Example:
+# username: Array (1 element)
+#   [0]: 'al-ben
+# repositories: Array (4 elements)
+#   [0]: 'repo1'
+#   [1]: 'repo2'
+def create_field_array(issue_response)
+    Hash[issue_response
+            # Fetch the body var and split on field seperator
+            .fetch("body").split("###")
+            # Remove empty lines
+            .map { |line| line.strip }.map { |str| str.gsub /^$\n/, '' }
+            # Split on \n characters to have field name and field value in seperator hash
+            .map { |line| line.split("\n") }
+            # Drop first hash element as not needed
+            .drop(1)
+            # Map values into array
+            .map { |field| [field[0], field.drop(1)] }]
+end
+
+# Grab needed data from GitHub Actions payload
+issue_json = JSON.parse(ENV.fetch("ISSUE"))
+
+# Grab issue fields from GitHub Actions payload
+fields = create_field_array(issue_json)
+
+# For each repo
```

Comment:
I would pull the actual tf editing into its own class/file so that we can have some basic unit tests around it.

Reply:
Have created a class for it